nni/compression/experiment/config/compression.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import List, Type
+
+from torch.nn import Module
+
+from nni.experiment.config import ExperimentConfig
+from nni.experiment.config.base import ConfigBase
+from .pruner import PrunerConfig
+from .quantizer import QuantizerConfig
+from .common import ComparableType
+
+__all__ = ['CompressionConfig', 'CompressionExperimentConfig']
+
+
+@dataclass
+class CompressionConfig(ConfigBase):
+    """
+    Attributes
+    ----------
+    params
+        The upper remaining ratio bound of the model params.
+    flops
+        The upper remaining ratio bound of the model flops.
+    metric
+        The lower remaining ratio bound of the model metric.
```

Comment:
what is the meaning of "metric", how do users define metric?

Reply:
add docstr
```
    metric
        The lower bound of the ratio of remaining model metric.
        Metric is the evaluator's return value, usually it is a float number representing the model accuracy.
        E.g., 0.9 means the compressed model should have at least 90% of the performance compared to the original model.
        This means that if the accuracy of the original model is 80%, then the accuracy of the compressed model should
        not be lower than 72% (0.9 * 80%).
```